File: python/Program/Garage.py
#Saya Muhammad Yusdan Ali Batubara NIM 2206847 mengerjakan latihan 4 dalam mata 
#kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya 
#tidak melakukan kecurangan seperti yang telah dispesifikasikan Aamiin.

class Garage:
    def __init__(self, nama_garasi, luas_garasi):
        self.nama_garasi = nama_garasi
        self.luas_garasi = luas_garasi
        self.list_kendaraan = []

    def setNamaGarasi(self, nama_garasi):
        self.nama_garasi = nama_garasi

    def getNamaGarasi(self):
        return self.nama_garasi
    
    def setLuasGarasi(self, luas_garasi):
        self.luas_garasi = luas_garasi

    def getLuasGarasi(self):
        return self.luas_garasi
    
    def addKendaraan(self, kendaraan):
        self.list_kendaraan.append(kendaraan)
       

File: cpp/Garage.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

// Import class Vehicle
#include "Vehicle.cpp" // Assume Vehicle class is defined in Vehicle.h

class Garage {
private:
    string nama_garasi;
    double luas_garasi;
    vector<Vehicle> list_kendaraan;

public:
    // Constructor
    Garage(string nama_garasi, double luas_garasi) {
        this->nama_garasi = nama_garasi;
        this->luas_garasi = luas_garasi;
    }

    // Setters
    void setNamaGarasi(string nama_garasi) {
        this->nama_garasi = nama_garasi;
    }

    void setLuasGarasi(double luas_garasi) {
        this->luas_garasi = luas_garasi;
    }

    // Getters
    string getNamaGarasi() {
        return this->nama_garasi;
    }

    double getLuasGarasi() {
        return this->luas_garasi;
    }

    // Method to add vehicle to garage
    void addKendaraan(Vehicle kendaraan) {
        this->list_kendaraan.push_back(kendaraan);
    }

    //getListKendaraan
    vector<Vehicle> getListKendaraan() {
        return this->list_kendaraan;
    }
};